### PR TITLE
Fix error checking of optional parameter `storage-engine-cache-size`

### LINKED ISSUE #1986: Remove redundant config checking for mongodb container
* BigchainDB version: master
* Operating System: N/A
* Deployment Type: `Docker/Kubernetes`

### Description
The bigchaindb/mongodb container, checks for `storage_engine_cache_size` twice in the entrypoint script. In one of those instances, it tries to make sure `STORANGE_ENGINE_CACHE_SIZE` is always set otherwise it exits. This should not be the case, instead this parameter should be optional.

```
# sanity checks
if [[ -z "${REPLICA_SET_NAME:?REPLICA_SET_NAME not specified. Exiting!}" || \
    -z "${MONGODB_PORT:?MONGODB_PORT not specified. Exiting!}" || \
    -z "${MONGODB_FQDN:?MONGODB_FQDN not specified. Exiting!}" || \
    -z "${MONGODB_IP:?MONGODB_IP not specified. Exiting!}" || \
    -z "${MONGODB_KEY_FILE_PATH:?MONGODB_KEY_FILE_PATH not specified. Exiting!}" || \
    -z "${MONGODB_CA_FILE_PATH:?MONGODB_CA_FILE_PATH not specified. Exiting!}" || \
    -z "${MONGODB_CRL_FILE_PATH:?MONGODB_CRL_FILE_PATH not specified. Exiting!}" || \
    -z ${STORAGE_ENGINE_CACHE_SIZE:=''} ]] ; then
    #-z "${MONGODB_KEY_FILE_PASSWORD:?MongoDB Key File Password not specified. Exiting!}" || \
  exit 1
...
```
The above snippet, assigns a default value to `STORAGE_ENGINE_CACHE_SIZE` but if we use the default value the next snippet will exit the script.

```
if [ ! -z "$STORAGE_ENGINE_CACHE_SIZE" ]; then
  if [[ "$STORAGE_ENGINE_CACHE_SIZE" =~ ^[0-9]+(G|M|T)B$ ]]; then
    sed -i.bk "s|STORAGE_ENGINE_CACHE_SIZE|${STORAGE_ENGINE_CACHE_SIZE}|g" ${MONGODB_CONF_FILE_PATH}
  else
    echo "Invalid Value for storage engine cache size $STORAGE_ENGINE_CACHE_SIZE"
    exit 1
  fi
else
  sed -i.bk "/cache_size=/d" ${MONGODB_CONF_FILE_PATH}
fi
...
```

Solution is to remove the sanity check.

### PR DESCRIPTION
**Pull request by muawiakh**
## Description
- Keep the parameter optional
- Remove redundant checking of the same parameter
- Do not exit if the parameter is not specified

## Issues This PR Fixes
Fixes #1986

### CODE REVIEW

**Comment by ttmc** [comment]
It's worth noting that the careful re-examination of this particular code was prompted by @r7vme in PR #1984

**Comment by ttmc** [comment]
This PR also updates the tag (version number) of various MongoDB Docker images.

File: k8s/mongodb/container/mongod_entrypoint.bash [hunk 1]
```diff
@@ -65,9 +65,8 @@ if [[ -z "${REPLICA_SET_NAME:?REPLICA_SET_NAME not specified. Exiting!}" || \
     -z "${MONGODB_IP:?MONGODB_IP not specified. Exiting!}" || \
     -z "${MONGODB_KEY_FILE_PATH:?MONGODB_KEY_FILE_PATH not specified. Exiting!}" || \
     -z "${MONGODB_CA_FILE_PATH:?MONGODB_CA_FILE_PATH not specified. Exiting!}" || \
-    -z "${MONGODB_CRL_FILE_PATH:?MONGODB_CRL_FILE_PATH not specified. Exiting!}" || \
-    -z ${STORAGE_ENGINE_CACHE_SIZE:=''} ]] ; then
-    #-z "${MONGODB_KEY_FILE_PASSWORD:?MongoDB Key File Password not specified. Exiting!}" || \
+    -z "${MONGODB_CRL_FILE_PATH:?MONGODB_CRL_FILE_PATH not specified. Exiting!}" ]] ; then
```
Comment (ttmc) [inline k8s/mongodb/container/mongod_entrypoint.bash]: Maybe you could include a comment line so a careful reader can be reassured that you purposefully didn't include `STORAGE_ENGINE_CACHE_SIZE` (or others?). Maybe even say why.

Comment (muawiakh) [reply k8s/mongodb/container/mongod_entrypoint.bash]: Done

Comment (ttmc) [reply k8s/mongodb/container/mongod_entrypoint.bash]: :+1:

**Pull request merged**